ext/afform/core/Civi/Api4/Action/Afform/Submit.php

```diff
@@ -353,6 +353,9 @@ public static function processGenericEntity(AfformSubmitEvent $event) {
         // What to do here? Sometimes we should silently ignore errors, e.g. an optional entity
         // intentionally left blank. Other times it's a real error the user should know about.
         \Civi::log('afform')->debug('Silently ignoring exception in Afform processGenericEntity call for "' . $event->getEntityName() . '". Message: ' . $e->getMessage());
+        if ('mandatory_missing' === $e->getErrorCode()) {
```

Comment:
```suggestion
        if (1 === count($event->records) && 'mandatory_missing' === $e->getErrorCode()) {
```
This would'nt be a problem if we were still in times when a spouse wasn't optional for a household :wink:

Ok, this was obviously too optimistic. I think there should be a configurable option for making entities mandatory in *FormBuilder*, or at least consider the only entity in a form mandatory by default - what do you think?

Reply:
Typically that's done by making fields required. For example, if you make the spouse first/last name fields required, then effectively the form requires a spouse. Otherwise, if all fields are optional, then the entity is not required.

Reply:
Hm, but this is reaching limits in case you want to require a spouse but not a specific attribute (let's say either first/last name or e-mail should be required). Marking the entire entity required would leave the validation of mandatory attributes to the API and allow *FormBuilder* to display the `mandatory_missing` error message.

Would you agree that a form with a single entity (`count($event->records) === 1`) would justify `mandatory_missing` errors to be displayed?